### PR TITLE
add support for dynamodb_table

### PR DESCRIPTION
**Pull request by moadibfr**
| Q                 | A
| ----------------- | ---
| 🐛 Bug fix?       | no
| 🚀 New feature?   | yes
| ⚠ Deprecations?   | no
| ❌ BC Break       | no
| 🔗 Related issues | #210 
| ❓ Documentation  | yes

## Description

add support for dynamodb_table

### CODE REVIEW

File: pkg/resource/aws/aws_dynamodb_table_ext.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package aws
+
+import "github.com/cloudskiff/driftctl/pkg/resource"
+
+func (r *AwsDynamodbTable) NormalizeForState() (resource.Resource, error) {
+	// set default to empty struct for timeouts
+	if r.Timeouts == nil {
```
Comment (eliecharra) [inline pkg/resource/aws/aws_dynamodb_table_ext.go:7]: It's a Terraform only related field, for consistency purpose it's more straightforward to handle it like in other resource by adding a `diff:"-"` annotation.

File: pkg/remote/resource_enumeration_error_handler.go [hunk 2]
```diff
@@ -21,9 +21,9 @@ func HandleResourceEnumerationError(err error, alertr *alerter.Alerter) error {
 		return err
 	}
 
-	if reqerr.StatusCode() == 403 {
+	if reqerr.StatusCode() == 403 || strings.Contains(reqerr.Code(), "AccessDenied") {
 		message := fmt.Sprintf("Ignoring %s from drift calculation: Listing %s is forbidden.", listError.SupplierType(), listError.ListedTypeError())
-		logrus.Debugf(message)
+		//logrus.Debugf(message)
```
Comment (eliecharra) [inline pkg/remote/resource_enumeration_error_handler.go:26]: You can remove this line 😉

File: pkg/remote/resource_enumeration_error_handler.go [hunk 3]
```diff
@@ -21,9 +21,9 @@ func HandleResourceEnumerationError(err error, alertr *alerter.Alerter) error {
 		return err
 	}
 
-	if reqerr.StatusCode() == 403 {
+	if reqerr.StatusCode() == 403 || strings.Contains(reqerr.Code(), "AccessDenied") {
```
Comment (eliecharra) [inline pkg/remote/resource_enumeration_error_handler.go:24]: To be sure to catch only this case maybe we could check the status code too ?

```go
if reqerr.StatusCode() == 403 || (reqerr.StatusCode() == 400 && reqerr.Code() == "AccessDeniedException") {
```

File: pkg/remote/aws/test/dynamodb_table_list/results.golden.json [hunk 4]
```diff
@@ -0,0 +1 @@
+[]
```
Comment (eliecharra) [inline pkg/remote/aws/test/dynamodb_table_list/results.golden.json:1]: We expect two tables there ⚠️

**Review by eliecharra — CHANGES_REQUESTED** [review]

File: pkg/remote/resource_enumeration_error_handler.go (hunk 3 above)
Comment (moadibfr) [reply pkg/remote/resource_enumeration_error_handler.go:24]: From what we saw AccessDeniedException is always access denied it does not depends on the http code so I remove te 400 check but no matter what we do we won't catch everything, 403 on dynamo is still another error.

File: pkg/remote/aws/test/dynamodb_table_list/results.golden.json (hunk 4 above)
Comment (moadibfr) [reply pkg/remote/aws/test/dynamodb_table_list/results.golden.json:1]: no we don't it's the test for enumeration (list) error ^^

File: pkg/remote/resource_enumeration_error_handler.go (hunk 2 above)
Comment (moadibfr) [reply pkg/remote/resource_enumeration_error_handler.go:26]: :facepalm:

File: pkg/resource/aws/aws_dynamodb_table_ext.go (hunk 1 above)
Comment (moadibfr) [reply pkg/resource/aws/aws_dynamodb_table_ext.go:7]: ho you're right I thought these were real timeout not only used by tf...

File: pkg/remote/aws/test/dynamodb_table_list/results.golden.json (hunk 4 above)
Comment (moadibfr) [reply pkg/remote/aws/test/dynamodb_table_list/results.golden.json:1]: I could rename the test directory but I used the same naming for all the listing test...

Comment (eliecharra) [reply pkg/remote/aws/test/dynamodb_table_list/results.golden.json:1]: Yeah the naming is pretty confusing but we'll refacto this in another PR

**Review by wbeuil — APPROVED** [review]
Seems good !

**Review by eliecharra — APPROVED** [review]

**Pull request merged**